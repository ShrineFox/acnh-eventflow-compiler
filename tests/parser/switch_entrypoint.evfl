flow Test():
    switch TestActor1.Query0():
        case 0, 1:
entrypoint EP1:
            TestActor1.Action0()
        case 2:
            TestActor2.Action0()
entrypoint EP2:
            TestActor2.Action0()
entrypoint EP3:
    TestActor1.Action0()

