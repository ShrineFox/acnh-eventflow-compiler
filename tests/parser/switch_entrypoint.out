RootNode[name=Test, vardefs=[], out_edges=[Event8]]
ActionNode[name=Event0, action=EventFlowActionAction0(), params={}, out_edges=[Event10]]
ActionNode[name=Event3, action=EventFlowActionAction0(), params={}, out_edges=[Event5]]
ActionNode[name=Event5, action=EventFlowActionAction0(), params={}, out_edges=[Event10]]
SwitchNode[name=Event8, query=EventFlowQueryQuery0(), params={}, cases={'Event0': [0, 1], 'Event3': [2]}, out_edges=[Event0, Event3]]
ActionNode[name=Event10, action=EventFlowActionAction0(), params={}, out_edges=[]]

